src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -105,18 +107,68 @@ class ComService(
     val enrichedProbationSearchResults = entityProbationSearchResult.mapNotNull {
       val licences =
         licenceRepository.findAllByCrnAndStatusCodeIn(it.identifiers.crn, LicenceStatus.IN_FLIGHT_LICENCES)
-      val prisonerSearchResult = this.prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(it.identifiers.noms))
 
-      // If an empty list has been returned, there are no relevant licences relating to search for the offender
+      val comName =
+        if (it.manager.name?.forename.isNullOrBlank() || it.manager.name?.surname.isNullOrBlank()) {
+          null
+        } else {
+          "${it.manager.name?.forename} ${it.manager.name?.surname}".convertToTitleCase()
+        }
+
+      /**
+       *  If an empty list has been returned, we do not have a licence for the person currently but still need to return
+       *  a result for the PP as they may want to create a licence for them
+       */
       if (licences.isEmpty()) {
-        null
+        val isNomisNumberPopulated = !it.identifiers.noms.isNullOrBlank()
+
+        if (isNomisNumberPopulated) {
+          val prisonerSearchResults = this.prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(it.identifiers.noms))
+
+          if (prisonerSearchResults.isEmpty()) {
+            null
+          } else {
+            val prisonerSearchResult = prisonerSearchResults.first()
+
+            val licenceType = if (prisonerSearchResult.licenceExpiryDate.isNullOrBlank()) {
```

Comment:
This logic is mirrored from the logic to ascertain a licence in the frontend - can someone just double check my logic here - current logic can be found [here](https://github.com/ministryofjustice/create-and-vary-a-licence/blob/f0b53be71cef78d6aaf66262f2e08dba69aeb145/server/services/licenceService.ts#L596).